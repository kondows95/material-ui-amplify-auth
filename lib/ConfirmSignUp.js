"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var core_1 = require("@material-ui/core");
var react_intl_1 = require("react-intl");
var FormTitle_1 = __importDefault(require("./FormTitle"));
var material_ui_basic_parts_1 = require("material-ui-basic-parts");
var Password_1 = __importDefault(require("./Password"));
var ConfirmSignUp = function (props) {
    var formatMessage = react_intl_1.useIntl().formatMessage;
    var _a = react_1.default.useState({ email: '', password: '', confirmationCode: '' }), form = _a[0], setForm = _a[1];
    var handleChangeValue = function (fieldName) { return function (event) {
        var newForm = __assign({}, form);
        newForm[fieldName] = event.target.value;
        setForm(newForm);
    }; };
    var handleSubmit = function (event) {
        event.preventDefault();
        props.confirmSignUp(props.email, form['confirmationCode']);
    };
    var msgId = 'MUAA.ConfirmSignUp';
    var msgIdTitle = msgId + '.Title';
    var msgIdCode = msgId + '.Code';
    var msgIdPassword = msgId + '.Password';
    var msgIdSubmit = msgId + '.Submit';
    var content = (react_1.default.createElement(core_1.Container, { "data-testid": msgId, component: "main", maxWidth: "xs" },
        react_1.default.createElement("form", { onSubmit: handleSubmit },
            react_1.default.createElement(core_1.Box, { display: "flex", flexDirection: "column", mt: 8 },
                react_1.default.createElement(FormTitle_1.default, { "data-testid": msgIdTitle, label: formatMessage({ id: msgIdTitle, defaultMessage: 'Please confirm your email' }) }),
                react_1.default.createElement(core_1.Box, { display: "flex", justifyContent: "center", fontWeight: 600, color: "error.main" }, props.error),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(core_1.TextField, { label: formatMessage({ id: msgIdCode, defaultMessage: 'Confirmation Code' }), onChange: handleChangeValue('confirmationCode'), value: form.confirmationCode, variant: "outlined", required: true, fullWidth: true, inputProps: { 'data-testid': msgIdCode } })),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(Password_1.default, { label: formatMessage({ id: msgIdPassword, defaultMessage: 'New Password' }), testId: msgIdPassword, value: form.password, onChange: handleChangeValue('password') })),
                react_1.default.createElement(core_1.Box, { width: "100%", mt: 4, mb: 2, className: "relative" },
                    react_1.default.createElement(material_ui_basic_parts_1.MyButton, { type: "submit", label: formatMessage({ id: msgIdSubmit, defaultMessage: 'Submit' }), color: "primary", loading: props.loading, "data-testid": msgIdSubmit })),
                react_1.default.createElement(core_1.Box, { width: "100%", my: 2 },
                    react_1.default.createElement(core_1.Link, { href: "#", variant: "body2", onClick: handleSubmit },
                        "Resend code to ",
                        props.email))))));
    return props.authState === 'confirmSignUp' ? content : null;
};
exports.default = ConfirmSignUp;
