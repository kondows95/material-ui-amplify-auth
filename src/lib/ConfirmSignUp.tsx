import React, { FormEvent, ChangeEvent } from 'react';
import { Box, Container, TextField, Link } from '@material-ui/core';
import { useIntl } from 'react-intl';
import FormTitle from './FormTitle';
import { MyButton } from 'material-ui-basic-parts';

type Props = {
    authState: string | null;
    changeAuthState: (state: string) => void;
    confirmSignUp: (email: string, confirmationCode: string) => void;
    loading: boolean;
    error: string;
    email: string;
};

const ConfirmSignUp: React.FC<Props> = (props) => {
    const { formatMessage } = useIntl();

    type Form = { [key: string]: string };
    const [form, setForm] = React.useState<Form>({ email: '', password: '', confirmationCode: '' });

    type FormKey = 'email' | 'password' | 'confirmationCode';
    const handleChangeValue = (fieldName: FormKey) => (event: ChangeEvent<HTMLInputElement>): void => {
        const newForm = { ...form };
        newForm[fieldName] = event.target.value;
        setForm(newForm);
    };

    const handleSubmit = (event: FormEvent): void => {
        event.preventDefault();
        props.confirmSignUp(props.email, form['confirmationCode']);
    };

    const msgId = 'MUAA.ConfirmSignUp';
    const msgIdTitle = msgId + '.Title';
    const msgIdCode = msgId + '.Code';
    const msgIdPassword = msgId + '.Password';
    const msgIdSubmit = msgId + '.Submit';
    const content = (
        <Container data-testid={msgId} component="main" maxWidth="xs">
            <form onSubmit={handleSubmit}>
                <Box display="flex" flexDirection="column" mt={8}>
                    <FormTitle
                        data-testid={msgIdTitle}
                        label={formatMessage({ id: msgIdTitle, defaultMessage: 'Please confirm your email' })}
                    />
                    <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
                        {props.error}
                    </Box>
                    <Box width="100%" my={2}>
                        <TextField
                            label={formatMessage({ id: msgIdCode, defaultMessage: 'Confirmation Code' })}
                            onChange={handleChangeValue('confirmationCode')}
                            value={form.confirmationCode}
                            variant="outlined"
                            required
                            fullWidth
                            inputProps={{ 'data-testid': msgIdCode }}
                        />
                    </Box>

                    <Box width="100%" my={2}>
                        <TextField
                            label={formatMessage({ id: msgIdPassword, defaultMessage: 'New Password' })}
                            type="password"
                            autoComplete="new-password"
                            onChange={handleChangeValue('password')}
                            value={form.password}
                            variant="outlined"
                            required
                            fullWidth
                            inputProps={{ 'data-testid': msgIdPassword }}
                        />
                    </Box>

                    <Box width="100%" mt={4} mb={2} className="relative">
                        <MyButton
                            type="submit"
                            label={formatMessage({ id: msgIdSubmit, defaultMessage: 'Submit' })}
                            color="primary"
                            loading={props.loading}
                            data-testid={msgIdSubmit}
                        />
                    </Box>

                    <Box width="100%" my={2}>
                        <Link href="#" variant="body2" onClick={handleSubmit}>
                            Resend code to {props.email}
                        </Link>
                    </Box>
                </Box>
            </form>
        </Container>
    );

    return props.authState === 'confirmSignUp' ? content : null;
};

export default ConfirmSignUp;
