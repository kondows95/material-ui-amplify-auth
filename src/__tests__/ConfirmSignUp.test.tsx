import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import ConfirmSignUp from '../lib/ConfirmSignUp';

let tLib: TestLib;
let changeAuthState: jest.Mock;
let confirmSignUp: jest.Mock;
let resendSignUp: jest.Mock;

beforeEach(() => {
    changeAuthState = jest.fn();
    confirmSignUp = jest.fn();
    resendSignUp = jest.fn();
});

describe('Basic Usage', () => {
    const msgs = {
        'MUAA.ConfirmSignUp.Title': '#MUAA.ConfirmSignUp.Title',
        'MUAA.ConfirmSignUp.Code': '#MUAA.ConfirmSignUp.Code',
        'MUAA.ConfirmSignUp.Password': '#MUAA.ConfirmSignUp.Password',
        'MUAA.ConfirmSignUp.Submit': '#MUAA.ConfirmSignUp.Submit',
        'MUAA.ConfirmSignUp.ResendCode': '#MUAA.ConfirmSignUp.ResendCode',
    };
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en" messages={msgs}>
                    <ConfirmSignUp
                        authState={'confirmSignUp'}
                        changeAuthState={changeAuthState}
                        confirmSignUp={confirmSignUp}
                        resendSignUp={resendSignUp}
                        loading={false}
                        error={'MyError'}
                        email={'test@example.com'}
                    />
                </IntlProvider>
            )
        );
    });

    it('Click submit button', () => {
        tLib.changeValue('MUAA.ConfirmSignUp.Code', '12345');
        tLib.changeValue('MUAA.ConfirmSignUp.Password', 'myPassword');
        tLib.click('MUAA.ConfirmSignUp.Submit');
        expect(confirmSignUp).toHaveBeenCalledWith('test@example.com', '12345');
    });

    it('You can resend code', () => {
        tLib.click('MUAA.ConfirmSignUp.ResendCode');
        expect(resendSignUp).toHaveBeenCalledWith('test@example.com');
    });

    it('You can set label by Intl', () => {
        const html = tLib.render.container.innerHTML;
        expect(html).toContain('#MUAA.ConfirmSignUp.Title');
        expect(html).toContain('#MUAA.ConfirmSignUp.Code');
        expect(html).toContain('#MUAA.ConfirmSignUp.Password');
        expect(html).toContain('#MUAA.ConfirmSignUp.Submit');
        expect(html).toContain('#MUAA.ConfirmSignUp.ResendCode');
    });

    it('You can set error message', () => {
        expect(tLib.render.container.innerHTML).toContain('MyError');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});

it('Defalt Intl labels', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <ConfirmSignUp
                    authState={'confirmSignUp'}
                    changeAuthState={changeAuthState}
                    confirmSignUp={confirmSignUp}
                    resendSignUp={resendSignUp}
                    loading={false}
                    error={''}
                    email={'test@example.com'}
                />
            </IntlProvider>
        )
    );
    const html = tLib.render.container.innerHTML;
    expect(html).toContain('Please confirm your email');
    expect(html).toContain('Confirmation Code');
    expect(html).toContain('New Password');
    expect(html).toContain('Submit');
    expect(html).toContain('Resend code to test@example.com');
});

it('If authState is not forgotPasswordReset, return empty.', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <ConfirmSignUp
                    authState={'forgotPassword'}
                    changeAuthState={changeAuthState}
                    confirmSignUp={confirmSignUp}
                    resendSignUp={resendSignUp}
                    loading={false}
                    error={''}
                    email={'test@example.com'}
                />
            </IntlProvider>
        )
    );
    expect(tLib.render.container.textContent).toBe('');
});
