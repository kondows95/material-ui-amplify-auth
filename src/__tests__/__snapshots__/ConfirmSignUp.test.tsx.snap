// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
    data-testid="MUAA.ConfirmSignUp"
  >
    <form>
      <div
        class="MuiBox-root MuiBox-root-517"
      >
        <div
          class="MuiBox-root MuiBox-root-518"
          data-testid="MUAA.ConfirmSignUp.Title"
        >
          <div
            class="MuiBox-root MuiBox-root-519"
          >
            <div
              class="MuiBox-root MuiBox-root-520"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-530"
          >
            #MUAA.ConfirmSignUp.Title
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-531"
        >
          MyError
        </div>
        <div
          class="MuiBox-root MuiBox-root-532"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              #MUAA.ConfirmSignUp.Code
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="MUAA.ConfirmSignUp.Code"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-591 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-593"
                >
                  <span>
                    #MUAA.ConfirmSignUp.Code *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-595"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              #MUAA.ConfirmSignUp.Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                autocomplete="new-password"
                class="MuiInputBase-input MuiOutlinedInput-input"
                data-testid="MUAA.ConfirmSignUp.Password"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-591 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-593"
                >
                  <span>
                    #MUAA.ConfirmSignUp.Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-596 relative"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            data-testid="MUAA.ConfirmSignUp.Submit"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              #MUAA.ConfirmSignUp.Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root MuiBox-root-630"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
            href="#"
          >
            Resend code to test@example.com
          </a>
        </div>
      </div>
    </form>
  </main>
</DocumentFragment>
`;
